fenix/app/src/test/java/org/mozilla/fenix/translations/TranslationsDialogBindingTest.kt

```diff
@@ -0,0 +1,278 @@
+import mozilla.components.concept.engine.translate.DetectedLanguages
+import mozilla.components.concept.engine.translate.Language
+import mozilla.components.concept.engine.translate.TranslationEngineState
+import mozilla.components.concept.engine.translate.TranslationError
+import mozilla.components.concept.engine.translate.TranslationOperation
+import mozilla.components.concept.engine.translate.TranslationPair
+import mozilla.components.concept.engine.translate.TranslationSupport
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.components.support.test.rule.runTestOnMain
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.mozilla.fenix.R
+
+@RunWith(AndroidJUnit4::class)
+class TranslationsDialogBindingTest {
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    lateinit var browserStore: BrowserStore
+    private lateinit var translationsDialogStore: TranslationsDialogStore
+
+    private val tabId = "1"
+    private val tab = createTab(url = tabId, id = tabId)
+
+    @Test
+    fun `WHEN fromLanguage and toLanguage get updated in the browserStore THEN translations dialog actions dispatched with the update`() =
+        runTestOnMain {
+            val englishLanguage = Language("en", "English")
+            val spanishLanguage = Language("es", "Spanish")
+            translationsDialogStore = spy(TranslationsDialogStore(TranslationsDialogState()))
+            browserStore = BrowserStore(
+                BrowserState(
+                    tabs = listOf(tab),
+                    selectedTabId = tabId,
+                ),
+            )
+
+            val binding = TranslationsDialogBinding(
+                browserStore = browserStore,
+                translationsDialogStore = translationsDialogStore,
+                sessionId = tabId,
+                context = testContext,
+            )
+            binding.start()
+
+            val detectedLanguages = DetectedLanguages(
+                documentLangTag = englishLanguage.code,
+                supportedDocumentLang = true,
+                userPreferredLangTag = spanishLanguage.code,
+            )
+
+            val translationEngineState = TranslationEngineState(
+                detectedLanguages = detectedLanguages,
+                error = null,
+                isEngineReady = true,
+                requestedTranslationPair = TranslationPair(
+                    fromLanguage = englishLanguage.code,
+                    toLanguage = spanishLanguage.code,
+                ),
+            )
+
+            val supportLanguages = TranslationSupport(
+                fromLanguages = listOf(englishLanguage),
+                toLanguages = listOf(spanishLanguage),
+            )
+
+            browserStore.dispatch(
```

Comment:
I need it because  
val toSelected = translationsState.translationEngineState?.initialToLanguage(
                        translationsState.supportedLanguages?.toLanguages,
                    )
And the UpdateFromSelectedLanguage action gets called only when toSelected is not null.

Reply:
Thanks for the explanation!